ambry-network/src/main/java/com.github.ambry.network/http2/Http2ClientResponseHandler.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2020 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License, version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.github.ambry.network.http2;
+
+import com.github.ambry.network.RequestInfo;
+import com.github.ambry.network.ResponseInfo;
+import com.github.ambry.utils.SystemTime;
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelHandler;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.SimpleChannelInboundHandler;
+import io.netty.handler.codec.http.FullHttpResponse;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Process {@link io.netty.handler.codec.http.FullHttpResponse} translated from HTTP/2 frames
+ */
+@ChannelHandler.Sharable
+public class Http2ClientResponseHandler extends SimpleChannelInboundHandler<FullHttpResponse> {
+  protected final Logger logger = LoggerFactory.getLogger(getClass());
+  private Queue<ResponseInfo> responseInfoQueue0 = new ConcurrentLinkedQueue<>();
+  private Queue<ResponseInfo> responseInfoQueue1 = new ConcurrentLinkedQueue<>();
```

Comment:
what are the purpose of having two queues here?

Reply:
1. to reduce lock contention in this produce consumer problem. While `channelRead0` produces to one queue, `sendAndPoll` consumes the other one.
2. sendAndPoll() needs a stop while processing response. switching between two queue achieves this naturally.